### ISSUE TITLE
[CCA-Diversity] Move init questions to cycle-scripts

### ISSUE THREAD

**Issue by jvanbruegge** [cyclejs-community/create-cycle-app#101]
To keep things related to the flavor in the flavor and not in create-cycle-app

**Comment by nickbalestra** [comment]
I think is fair to have a better upfront DX with core-flavor.
For custom flavor yes.

**Comment by jvanbruegge** [comment]
would it change something? AFAIK, the questions would only be moved to the `init.js`

**Comment by nickbalestra** [comment]
Is already there. You mean removing it from the create-react-app?

**Comment by jvanbruegge** [comment]
I mean [this file](https://github.com/cyclejs-community/create-cycle-app/blob/master/packages/create-cycle-app/src/initQuestions.js)

**Comment by nickbalestra** [comment]
as in [this](https://github.com/cyclejs-community/create-cycle-app/blob/master/packages/cycle-scripts/scripts/init/questions.js) ?

**Comment by jvanbruegge** [comment]
yes ^^, so you just have to delete the other file?

**Comment by nickbalestra** [comment]
You would also want to remove the [check logic](https://github.com/cyclejs-community/create-cycle-app/blob/master/packages/create-cycle-app/src/createApp.js#L25-L33) if you want to do this. 

I was thinking of the guy who doesn't read docs (as everybody) just install and run the CLI, asking upfront will make it self-understandable of what is going to happen next, then braking this into the middle of various install. But I could be wrong ofc.

**Comment by jvanbruegge** [comment]
I dont think it matters if there is a small waiting period. You run cca without argumets, it downloads the cycle-scripts and then asks you what language and stream lib you want to use. Normal people usually wait if there is a progress bar indicating the program is doing something :)

**Comment by nickbalestra** [comment]
I'm fine with either ways... Though I think is important to advertise this change to people upfront. Perhaps after a while in the future we can think of remove it. But now people don't expect any question and they fact that the core flavor support multiple languages and stream library should be promoted imho. Afterward is not a big deal so we could definitely think of removing from cca

**Comment by jvanbruegge** [comment]
this has a massive downfall for us, we cant test locally. If you use the `--flavor` options the app throws an error, because `options` is not defined in `setup.js`

**Comment by nickbalestra** [comment]
That's why you need to handle it on both side. For testing this you can use the `--forceprompt` flag. Left you an [example](https://github.com/cyclejs-community/create-cycle-app/pull/102#issuecomment-292908978) on the PR. I'll plan to add integration tests for it on the cca, but we should also add unit and integration tests for the flavor as well.

**Comment by jvanbruegge** [comment]
Well not neccesary on both sides. Flavor side alone is enough :)

**Comment by nickbalestra** [comment]
Diversity release will focus on enable diversity. The prompt will be only on stream lib (xstream, rxjs, most) and such diversity API will be in charge of pass this information to the flavor. 

I suggest that as the CCA will be diversity, that prompt (streamLib) will be the only prompt at CCA level (not cycle-scripts). A flag to mute should be added.
Every other prompt is due to to flavor duties.
